src/Microsoft.ML.FastTree/TreeEnsemble/RegressionTree.cs

```diff
@@ -267,13 +279,15 @@ internal RegressionTree(ModelLoadContext ctx, bool usingDefaultValue, bool categ
                 if (categoricalNodeIndices?.Length > 0)
                 {
                     CategoricalSplitFeatures = new int[NumNodes][];
+                    CategoricalFeatureGain = new double[NumNodes][];
                     CategoricalSplitFeatureRanges = new int[NumNodes][];
                     foreach (var index in categoricalNodeIndices)
                     {
                         Contracts.Assert(CategoricalSplit[index]);
                         Contracts.Assert(index >= 0 && index < NumNodes);
 
                         CategoricalSplitFeatures[index] = reader.ReadIntArray();
+                        CategoricalFeatureGain[index] = reader.ReadDoubleArray();
```

Comment:
You've changed the format but didn't change the version number of the FastTree models, or account for the fact that existing models obviously won't have this information. Correspondingly, you've broken backwards compatibility.